src/routes/root.tsx

```diff
@@ -1,23 +1,64 @@
-import { data } from "@/data/sampledata";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import SearchBar from "@/components/SearchBar";
 import "@/css/Root.css";
 import DataTable from "@/components/DataTable";
 import PreviewCard from "@/components/PreviewCard";
+import { location } from "@/data/dataFilters";
 
 export default function Root() {
-  const [filteredData, setFilteredData] = useState<MCAPFileInformation[]>(data);
+  const [filteredData, setFilteredData] = useState<MCAPFileInformation[]>();
+  const [selectedRow, setSelectedRow] = useState<string>("");
+  const [selectedData, setSelectedData] = useState<MCAPFileInformation>();
+
+  useEffect(() => {
+    const fetchData = async (loc: string) => {
+      const res = await fetch(
+        `${process.env.API_URL}/api/v2/mcap/get?location=${loc}`,
+      );
+      const data = await res.json();
+
+      return data;
+    };
+
+    const assignData = async () => {
+      const apiData = await Promise.all(
+        location.map(async (loc: string) => await fetchData(loc)),
+      );
+      const validData = apiData.filter((data) => data !== null);
+
+      const flattenedData = validData.flat();
+
+      const sortedData = flattenedData.sort((a, b) => {
+        const [monthA, dayA, yearA] = a.date.split("-");
+        const [monthB, dayB, yearB] = b.date.split("-");
+        const dateA = new Date(`${yearA}-${monthA}-${dayA}`);
+        const dateB = new Date(`${yearB}-${monthB}-${dayB}`);
+
+        return dateB.getTime() - dateA.getTime();
+      });
+
+      setFilteredData(sortedData);
+    };
+
+    assignData();
+  }, []);
```

Comment:
when/how should this actually get called? the dependency array for the useEffect is empty meaning it only gets ran on page load?